.github/workflows/nix.yml

```diff
@@ -47,7 +47,7 @@ jobs:
         name: haskell-language-server
         authToken: ${{ secrets.HLS_CACHIX_AUTH_TOKEN }}
     - if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
-      run: nix build
+      run: nix build || (nix develop --command cabal update && nix develop --command cabal build)
```

Comment:
Can we split this in two jobs instead? One for the `nix build` and one for the `cabal build`. The first one would not be required to merge.